src/layouts/index.js

```diff
@@ -56,47 +56,48 @@ export default class DefaultLayout extends Component {
    *
    * Returns a map wherein each key is the footnote name, and each value is an object containing:
    * - labels: the numerical label for the footnote
-   * - references: a list of the ids that refer to this footnote
+   * - references: a list of the footnote reference ids that refer to this footnote
    */
   getFootnotes = nodes => {
     const footnotes = findAllKeyValuePairs(nodes, 'type', 'footnote');
+    const footnoteReferences = findAllKeyValuePairs(nodes, 'type', 'footnote_reference');
+    // Label our footnotes by their index, regardless of their names to
+    // circumvent cases such as [[1], [#], [2], ...]
     return footnotes.reduce((map, footnote, index) => {
-      // Track how many anonymous footnotes are on the page so that we can correctly associate footnotes and references
-      let anonymousCount = 0;
       if (footnote.name) {
         // Find references associated with a named footnote
         // eslint-disable-next-line no-param-reassign
         map[footnote.name] = {
           label: index + 1,
-          references: this.getNamedFootnoteReferences(nodes, footnote.name),
+          references: this.getNamedFootnoteReferences(footnoteReferences, footnote.name),
         };
       } else {
         // Find references associated with an anonymous footnote
+        // Replace potentially broken anonymous footnote ids by overwriting
+        // with their index
+        footnote.id = '' + (index + 1);
```

Comment:
Assuming the intent of the concatenation here was to make `footnote.id` into a string, a template literal may be clearer:
```suggestion
        footnote.id = `${index + 1}`;
```